src/werkzeug/formparser.py

```diff
@@ -461,6 +469,160 @@ def validate_boundary(self, boundary):
+                yield b""
+                if not input_buffer:
+                    yield leftover_buffer
+                else:
+                    input_buffer = leftover_buffer + input_buffer
+            else:
+                break
+
+    class LineParser:
+        def __init__(self, parent, boundary):
+            self.parent = parent
+            self.boundary = boundary
+
+            self._next_part = b"--" + boundary
+            self._last_part = self._next_part + b"--"
+            self._state_enum = ParseEnums.pre_term
+            self._states = {
+                ParseEnums.pre_term: self._state_pre_term,
+                ParseEnums.headers: self._state_headers,
+                ParseEnums.output: self._state_output,
+            }
+            self._headers = []
+            self._headers_only = False
+            self._tail = b""
+            self._codec = None
+
+        def _state_pre_term(self, line):
+            while True:
+                if not line:
+                    self.parent.fail("unexpected end of file")
+                else:
+                    line = line.rstrip(b"\r\n")
+
+                    if not line:
+                        yield None, ParseEnums.pre_term
+                    elif line == self._last_part:
+                        yield None, ParseEnums.done
+                    elif line == self._next_part:
+                        yield None, ParseEnums.headers
+                    else:
+                        self.parent.fail("expected boundary at start of multipart data")
+
+        def _state_headers(self, line):
+            while True:
+                if line is None:
+                    self.parent.fail("unexpected end of file during headers")
+
+                line = _to_str(line)
+                line, line_terminated = _line_parse(line)
+                if not line_terminated:
+                    self.parent.fail("unexpected end of line in multipart header")
+
+                if not line:
+                    self._headers = Headers(self._headers)
+                    if self._headers_only:
+                        yield self._headers, ParseEnums.done
+                    else:
+                        disposition = self._headers.get("content-disposition", None)
+                        if disposition is None:
+                            self.parent.fail("missing Content-Disposition header")
+                        else:
+                            disposition, extra = parse_options_header(disposition)
+                            transfer_encoding = self.parent.get_part_encoding(
+                                self._headers
+                            )
+                            if transfer_encoding is not None:
+                                if transfer_encoding == "base64":
+                                    transfer_encoding += "_codec"
+
+                            try:
+                                self._codec = codecs.lookup(transfer_encoding)
```

Comment:
What if `self._codec` has already been set?

Reply:
I am not sure how we can handle that. Could you elaborate more please?